javaslang/src/test/java/javaslang/collection/SetsEqualityTest.java

```diff
@@ -0,0 +1,73 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import javaslang.control.Option;
+import org.junit.Test;
+
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SetsEqualityTest {
+
+    private final String[] testStrings = { "a", "b", "c", "D" };
+
+    private final Map<Class<?>, SetStringGenerator> generators = HashMap.of(
+            HashSet.class, SetStringGenerator.hashSet,
+            LinkedHashSet.class, SetStringGenerator.linkedHashSet,
+            TreeSet.class, SetStringGenerator.treeSet);
+
+    @Test
+    public void shouldAllImplementationsBeEqualForTheSameData() {
+        pairwiseDistinctSetClasses().forEach(pair -> this.equalityTest(pair._1, pair._2));
+    }
```

Comment:
Please add tests for the negative case (sets are not equal). Also the second set should be initialized with test strings of different order (because of LinkedHashSet).